kobo/apps/organizations/tests/test_organization_invitations.py

```diff
@@ -63,57 +65,86 @@ def _update_invite(self, user, guid, status):
-            self.assertIn(
-                invitation['invitee'], self.invitation_data['invitees']
-            )
-            # Check that the email was sent
-            invite = User.objects.filter(
-                Q(username=invitation['invitee']) |
-                Q(email=invitation['invitee'])
-            ).first()
+        user = getattr(self, f'{user_role}_user')
+        response = self._create_invite(user)
+        self.assertEqual(response.status_code, expected_status)
+        if response.status_code == status.HTTP_201_CREATED:
             self.assertEqual(
-                mail.outbox[index].to[0],
-                invite.email if invite else invitation['invitee']
+                len(response.data), len(self.invitation_data['invitees'])
             )
+            for index, invitation in enumerate(response.data):
+                self.assertIn(
+                    invitation['invitee'], self.invitation_data['invitees']
+                )
+                # Check that the email was sent
+                invite = User.objects.filter(
+                    Q(username=invitation['invitee']) |
+                    Q(email=invitation['invitee'])
+                ).first()
+                self.assertEqual(
+                    mail.outbox[index].to[0],
+                    invite.email if invite else invitation['invitee']
+                )
 
-    def test_owner_can_resend_invitation(self):
+    @data(
+        ('owner', status.HTTP_200_OK),
+        ('admin', status.HTTP_200_OK),
+        ('member', status.HTTP_403_FORBIDDEN),
+        ('external', status.HTTP_403_FORBIDDEN)
+    )
+    @unpack
+    def test_owner_can_resend_invitation(self, user_role, expected_status):
         """
-        Test that the organization owner can resend an invitation
+        Test that only organization owner or admin can resend an invitation
         """
         self._create_invite(self.owner_user)
+        user = getattr(self, f'{user_role}_user')
+        self.client.force_login(user)
         invitation = OrganizationInvitation.objects.get(
             invitee=self.external_user
         )
         response = self.client.patch(
             self.detail_url(invitation.guid), data={'status': 'resent'}
         )
-        self.assertEqual(response.status_code, status.HTTP_200_OK)
-        self.assertEqual(response.data['status'], 'resent')
-        self.assertEqual(mail.outbox[0].to[0], invitation.invitee.email)
+        self.assertEqual(response.status_code, expected_status)
+        if response.status_code == status.HTTP_200_OK:
+            self.assertEqual(response.data['status'], 'resent')
+            self.assertEqual(mail.outbox[0].to[0], invitation.invitee.email)
 
-    def test_owner_can_cancel_invitation(self):
+    @data(
+        ('owner', status.HTTP_200_OK),
+        ('admin', status.HTTP_200_OK),
+        ('member', status.HTTP_403_FORBIDDEN),
+        ('external', status.HTTP_403_FORBIDDEN)
```

Comment:
I think it should 404 for external. We don't want to reveal the existence of the organization to external.